plugins/tailwind/compiler.ts

```diff
@@ -76,3 +97,44 @@ export async function initTailwind(
 
   return postcss(plugins);
 }
+
+const fileTypes = [".tsx"];
+
+function extractSpecifiers(graph: ModuleGraphJson, projectLocation: string) {
+  const specifiers: Set<string> = new Set();
+
+  for (const module of graph.modules) {
+    if (
+      fileTypes.some((type) =>
+        module.specifier.endsWith(type) &&
+        module.specifier.startsWith(path.dirname(projectLocation))
+      )
+    ) {
+      specifiers.add(module.specifier);
+    }
+  }
+  return Array.from(specifiers);
+}
+
+function readDenoConfig() {
+  try {
+    const configText = Deno.readTextFileSync("./deno.json");
+    const config = JSON.parse(configText);
+    return config;
+  } catch (err) {
+    console.error("Error reading deno.json:", err);
+    return null;
+  }
+}
+
+function createCustomResolver(imports: Record<string, string>) {
+  return function customResolve(specifier: string, referrer: string): string {
+    for (const [key, value] of Object.entries(imports)) {
+      if (specifier.startsWith(key)) {
+        const mappedPath = specifier.replace(key, value);
+        return new URL(mappedPath, referrer).toString();
+      }
+    }
+    return new URL(specifier, referrer).toString();
+  };
+}
```

Comment:
```suggestion
function createCustomResolver(imports: Record<string, string>) {
  return (specifier: string, referrer: string): string => {
    const importMatch = findSingle(
      imports,
      ({ key }) => specifier.startsWith(key),
    );
    if (importMatch !== undefined) {
      specifier = specifier.replace(importMatch.key, importMatch.value);
    }
    return new URL(specifier, referrer).toString();
}
```
Perhaps, you could do something like this. See https://deno.land/std@0.212.0/collections/mod.ts?s=findSingle